lib/nats.js

```diff
@@ -1007,7 +1050,7 @@ Client.prototype.publish = function(subject, msg, opt_reply, opt_callback) {
   if (opt_callback !== undefined) {
     this.flush(opt_callback);
   } else if (this.closed) {
-    throw(new Error(CONN_CLOSED));
+    throw(new NatsError(CONN_CLOSED_MSG, CONN_CLOSED));
```

Comment:
Maybe out of scope for this PR, but also another idea is that instead of using a code to determine the error, new error types based could be created which extend from `NatsError` could be added, like a `ConnectionClosedError` for example.